generator/vertx/src/main/java/io/fabric8/maven/generator/vertx/VertxGenerator.java

```diff
@@ -35,13 +52,98 @@
         super(context, "vertx");
     }
 
     @Override
     public boolean isApplicable(List<ImageConfiguration> configs) throws MojoExecutionException {
-        // Disabled for now
+        return shouldAddImageConfiguration(configs)
+                && MavenUtil.hasPlugin(getProject(), VERTX_MAVEN_PLUGIN_GA);
+    }
+
+    @Override
+    protected boolean isFatJar() throws MojoExecutionException {
+        if (!hasMainClass() && isVertxPackage()) {
+            return true;
+        }
+        return super.isFatJar();
+    }
+
+
+    @Override
+    protected List<String> extractPorts() {
+        List<String> ports = portsFromConfig();
+        return ports != null ? ports : FALLBACK_PORTS;
+    }
+
+    protected List<String> portsFromConfig() {
+        List<String> answer = new ArrayList<>();
+        MavenProject project = getProject();
+        Plugin plugin = project.getPlugin(VERTX_MAVEN_PLUGIN_GA);
+        if (plugin != null) {
+            Xpp3Dom configuration = (Xpp3Dom) plugin.getConfiguration();
+            if (configuration == null) {
+                return null;
+            }
+            Xpp3Dom config = configuration.getChild("config");
+            String configPath = config != null ? config.getValue() : System.getProperty(VERTX_CONFIG, null);
+            if (configPath == null) {
+                return null;
+            }
+            File configFile = Paths.get(configPath).toFile();
+            if (!configFile.exists()) {
+                log.warn("Could not find config: [" + configPath + "]. Ignoring.");
+                return null;
+            }
+
+            try {
+                Map<String, String> configMap = OBJECT_MAPPER.readValue(configFile, Map.class);
+                for (Map.Entry<String, String> entry : configMap.entrySet()) {
+                    String key = entry.getKey();
+                    Matcher m = PORT_PATTERN.matcher(key);
+                    if (m.matches()) {
+                        addPortIfValid(answer, String.valueOf(entry.getValue()));
+                    }
+                }
+            } catch (IOException e) {
+                log.warn("Error reading config: [" + configPath + "], due to: [" + e.getMessage() + "]. Ignoring.");
+            }
+        }
+        return answer;
+    }
+
+    private void addPortIfValid(List<String> list, String port) {
+        if (Strings.isNotBlank(port)) {
+            list.add(port);
+        }
+    }
+
+    private boolean isVertxPackage() {
+        MavenProject project = getProject();
+        Plugin plugin = project.getPlugin(VERTX_MAVEN_PLUGIN_GA);
+        if (plugin != null) {
+            Map<String, PluginExecution> executionsAsMap = plugin.getExecutionsAsMap();
+            if (executionsAsMap != null) {
+                for (PluginExecution execution : executionsAsMap.values()) {
+                    List<String> goals = execution.getGoals();
+                    if (goals.contains("package")) {
```

Comment:
I wonder whether should be more open as the vertx-maven-plugin could also be used when no bound to an execution phase, e.g. when called directly like in `mvn package vertx:package fabric8:build`. I don't know whether it is possible but it would be awesome if it would be possible to find out which goals are actually in effect when this goal is reached. There is a `GeneratorContext.runningWithGoal()` but I don't know whether it really catches also "resolved" calls.